### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by WhiteSparrows** [unit8co/darts#1314]
**Describe the bug**
When trying to do a gridsearch using an early stopper, it returns an error : __deepcopy__() takes 1 positional argument but 2 were given.

This issue arise using RNNModel (LSTM) and NBEATS

**To Reproduce**
from darts.models import RNNModel
from pytorch_lightning.callbacks.early_stopping import EarlyStopping

my_stopper = EarlyStopping(
    monitor="train_loss",
    patience=5,
    min_delta=0.005,
    mode='min',
)
PL_TRAINER = {"callbacks": [my_stopper]}
params = {
            "input_chunk_length": [12, 52, 256, 512],
            "model": ['LSTM'],
            "hidden_dim": [5, 10, 20, 50],
            "dropout": [0.1],
            "n_rnn_layers": [3, 5],
            "batch_size": [16],
            "n_epochs": [20],
            "optimizer_kwargs":[{"lr": 1e-3}],
            "pl_trainer_kwargs": [PL_TRAINER],
            "nr_epochs_val_period": [1],
            "random_state": [0]
        }

# Randomized gridsearch
res = RNNModel.gridsearch(
            parameters=params,
            series=my_series,
            forecast_horizon=24*14,
            stride=24*14,
            past_covariates=covariates,
            start=0.5,
            metric=mse,
            reduction=np.mean,
            verbose=True,
        )

**Expected behavior**
Find the best combination of parameters after several training using an expanding windows mode

**System (please complete the following information):**
 - Python version: [e.g. 3.7]
 - darts version [e.g. 0.14.0]

**Additional context**
Early stopper worked perfectly with historical_forecast (code below)
Maybe link to this issue : https://github.com/unit8co/darts/issues/1261

my_stopper = EarlyStopping(
    monitor="train_loss",
    patience=5,
    min_delta=0.005,
    mode='min',
)

PL_TRAINER = {"callbacks": [my_stopper]}
INPUT = 52 # 7 * 24
HIDDEN_DIM = 20
DROPOUT = 0.1
RNN_LAYERS = 5
BATCH = 16
EPOCHS = 50
LR = 1e-3
LOSS_FN = MSELoss()
TRAINING_LENGTH = 512

lstm = RNNModel(
    model="LSTM",
    input_chunk_length= INPUT,
    hidden_dim=HIDDEN_DIM,
    dropout=DROPOUT,
    n_rnn_layers=RNN_LAYERS,
    batch_size=BATCH,
    n_epochs=EPOCHS,
    optimizer_kwargs={"lr": LR},
    model_name="LSTM",
    log_tensorboard=False,
    random_state=0,
    loss_fn = LOSS_FN,
    training_length=TRAINING_LENGTH,
    pl_trainer_kwargs=PL_TRAINER,
    force_reset=True,
    save_checkpoints=False,
)

pred_series_retrain = lstm.historical_forecasts(
                    ts_conso,
                    start=0.1,
                    past_covariates=covariates,
                    forecast_horizon=24*14,
                    stride=24*14, 
                    retrain=True,
                    overlap_end=False,
                    last_points_only=False,
                    verbose=True)
